src/lib.zig

```diff
@@ -52,16 +52,23 @@ pub fn serializedSize(comptime T: type, data: T) !usize {
         .bool => @sizeOf(T),
         .array => size: {
             var size: usize = 0;
+            const is_child_fixed = try isFixedSizeObject(info.array.child);
             for (0..data.len) |i| {
+                if (!is_child_fixed) {
+                    size += 4;
+                }
                 size += try serializedSize(info.array.child, data[i]);
             }
```

Comment:
Better add a single value once, than over and over again. The compiler might be able to do the same thing, but why take a chance?

```suggestion
for (0..data.len) |i| {
    size += try serializedSize(info.array.child, data[i]);
}
if (!is_child_fixed) {
    size += 4 * data.len;
}
```